sol/contracts/ics23Proof.sol

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.2;
+
+import {LeafOp, CompressedBatchProof, ExistenceProof, NonExistenceProof, BatchEntry, BatchProof, ProofSpec, InnerOp, InnerSpec, CommitmentProof} from "./proofs.sol";
+import {Ops} from "./ics23Ops.sol";
+import {SafeCast} from "OpenZeppelin/openzeppelin-contracts@4.2.0/contracts/utils/math/SafeCast.sol";
+import {BytesLib} from "GNSPS/solidity-bytes-utils@0.8.0/contracts/BytesLib.sol";
+import {Compress} from "./ics23Compress.sol";
+import {Ops} from "./ics23Ops.sol";
+
+library Proof{
+    bytes constant empty = new bytes(0);
+
+    enum VerifyExistenceError{
+        None,
+        KeyNotMatching,
+        ValueNotMatching,
+        CheckSpec,
+        CalculateRoot,
+        RootNotMatching
+    }
+    // ExistenceProof
```

Comment:
These library internal methods should all be documented with [NatSpec](https://docs.soliditylang.org/en/v0.8.2/natspec-format.html).